administrator/components/com_content/src/Model/ArticleModel.php

```diff
@@ -760,21 +758,38 @@ public function save($data)
 		// Alter the title for save as copy
 		if ($input->get('task') == 'save2copy')
 		{
-			$origTable = clone $this->getTable();
-			$origTable->load($input->getInt('id'));
+			$origTable = $this->getTable();
+
+			if (Factory::getApplication()->isClient('site'))
+			{
+				$id = $input->getInt('a_id');
+
+				// For frontend, alias is not entered, we set it to empty to have Alias generated base on title of article
+				$data['alias'] = '';
+			}
+			else
+			{
+				$id = $input->getInt('id');
+			}
+
+			$origTable->load($id);
 
 			if ($data['title'] == $origTable->title)
 			{
+				// If user did not change title before press Save as Copy in frontend, set Alias to original article
+				// alias to have the system generates Title and Alias automatically for the creating article
```

Comment:
```suggestion
				/**
				 * If user did not change title before press Save as Copy in frontend, set Alias to original article
				 * alias to have the system generates Title and Alias automatically for the creating article.
				 */
```

reformat to multi line comment - https://developer.joomla.org/coding-standards/inline-code-comments.html

Reply:
Can you write it in english as well please